python/ray/tests/test_client_multi.py

```diff
@@ -0,0 +1,173 @@
+import sys
+import pytest
+import ray
+
+
+@pytest.mark.skipif(
+    sys.platform == "win32",
+    reason="PSUtil does not work the same on windows.")
+@pytest.mark.parametrize(
+    "call_ray_start",
+    ["ray start --head --ray-client-server-port 25001 --port 0"],
+    indirect=True)
+def test_multi_cli_basic(call_ray_start):
+    cli1 = ray.init("ray://localhost:25001", allow_multiple=True)
+    cli2 = ray.init("ray://localhost:25001", allow_multiple=True)
+    with cli1:
+        a = ray.put(10)
+    with cli2:
+        b = ray.put(20)
+
+    with pytest.raises(Exception):
+        ray.put(30)
+
+    with pytest.raises(Exception):
+        ray.get(a)
+
+    with pytest.raises(Exception), cli2:
+        ray.get(a)
+
+    with pytest.raises(Exception), cli1:
+        ray.get(b)
+
+    ray.init("ray://localhost:25001")
+    c = ray.put(30)
+
+    with cli1:
+        assert 10 == ray.get(a)
+
+    with cli2:
+        assert 20 == ray.get(b)
+
+    with pytest.raises(Exception), cli1:
+        ray.get(c)
+
+    with pytest.raises(Exception), cli2:
+        ray.get(c)
+
+
+@pytest.mark.skipif(
+    sys.platform == "win32",
+    reason="PSUtil does not work the same on windows.")
+@pytest.mark.parametrize(
+    "call_ray_start",
+    ["ray start --head --ray-client-server-port 25001 --port 0"],
+    indirect=True)
+def test_multi_cli_init(call_ray_start):
+    cli1 = ray.init("ray://localhost:25001", allow_multiple=True)  # noqa
+    cli2 = ray.init("ray://localhost:25001")  # noqa
```

Comment:
Should this raise?

Reply:
@ericl I'm implementing it according to the rules we discussed in the comment.

ray.init(, allow_multiple=False)

=> if there is one connection

=> if it's connected with allow_multiple=True -> proceed.



Could you let me know what's the expected behavior here?

Reply:
Yep, this was a case I missed. It should raise here.